### PR TITLE
Add docs for frontend integration + opt-in for Jenkins integration

### PR DESCRIPTION
**Pull request by timja**
Fixes https://github.com/jenkinsci/plugin-pom/issues/874

Restores the ability to use Jenkins to report test failures (broken in https://github.com/jenkinsci/plugin-pom/pull/1008) and implements support for linting.

See also discussion in https://github.com/jenkinsci/plugin-pom/pull/1015#issuecomment-2368535480 where it was decided that the restoring of the support will be opt-in to ensure that by default plugins builds will not silently fail on frontend test / lint failures.

### Testing done

Tested with a local SNAPSHOT in https://github.com/jenkinsci/workflow-cps-plugin/pull/935

### Submitter checklist
- [ ] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [ ] Ensure that the pull request title represents the desired changelog entry
- [ ] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Review by jglick — APPROVED** [review]
Sounds OK. Is there a quick way to determine whether a given repo contains relevant tests, like a `test -d something`? Are there any other @jenkinsci plugins misconfigured? (There might be some @cloudbees plugins affected, so it would be nice to have a simple thing to search for in GitHub.)

**Comment by timja** [comment]
> Sounds OK. Is there a quick way to determine whether a given repo contains relevant tests, like a `test -d something`? Are there any other @jenkinsci plugins misconfigured? (There might be some @cloudbees plugins affected, so it would be nice to have a simple thing to search for in GitHub.)

This search brings up the plugins that could be affected:
https://github.com/search?q=org%3Ajenkinsci%20mvntest&type=code

Not sure of an easy way to tell if misconfigured as you can configure it in multiple different ways.
e.g. at first glance I thought this was misconfigured:
https://github.com/jenkinsci/active-choices-plugin/blob/6736e8c22d18cadd5635410589522cd85c2be333/package.json#L12

but if you look further down you'll see its been configured a different way:
https://github.com/jenkinsci/active-choices-plugin/blob/6736e8c22d18cadd5635410589522cd85c2be333/package.json#L44-L67

**Comment by timja** [comment]
Looking through the list further I saw a few categories:

1. empty string or `exit 0` defined for `mvntest`
2. jest with junit-jest setup
3. `jjsbuilder`
4. `gulp`
5. `eslint` - these are mostly misconfigured
  1. e.g. https://github.com/jenkinsci/workflow-cps-plugin/blob/0767b4ba0bfaff516d26190f151676b116cc6e61/plugin/package.json#L14
  2. https://github.com/jenkinsci/job-dsl-plugin/blob/616477fe247803b7d45f8be2d2e54b5f84ee0450/job-dsl-plugin/package.json#L8

For eslint there's an example in Jenkins core using the checkstyle format: https://github.com/jenkinsci/jenkins/blob/master/package.json#L18

I suspect anything using `jjsbuilder` or `gulp` will not be configured right and is likely unmaintained / very old.

**Comment by timja** [comment]
I'll add an eslint example before taking this out of draft and add it to workflow-cps

File: README.md [hunk 1]
```diff
@@ -156,6 +156,118 @@ By default, the setup wizard (Jenkins >= 2.0) is skipped when using `hpi:run`. I
...
+
+<!-- TODO when enough plugins have adopted this then remove this step -->
+Then set the following properties in your `pom.xml` to configure the Maven build to let Jenkins report the results:
+
+```xml
+<properties>
+  <maven.test.failure.ignore>false</maven.test.failure.ignore>
+  <testFailureIgnore>${maven.test.failure.ignore}</testFailureIgnore>
+</properties>
+```
```
Comment (jglick) [inline README.md:234]: https://github.com/jenkinsci/workflow-cps-plugin/pull/935#discussion_r1773433882

File: README.md [hunk 2]
```diff
@@ -156,6 +156,118 @@ By default, the setup wizard (Jenkins >= 2.0) is skipped when using `hpi:run`. I
...
+
+```
+
+<!-- TODO when enough plugins have adopted this then remove this step -->
+Then set the following properties in your `pom.xml` to configure the Maven build to let Jenkins report the results:
+
+```xml
+<properties>
+  <maven.test.failure.ignore>false</maven.test.failure.ignore>
+  <testFailureIgnore>${maven.test.failure.ignore}</testFailureIgnore>
```
Comment (timja) [inline README.md:232]:
```suggestion
  <frontend.testFailureIgnore>${maven.test.failure.ignore}</frontend.testFailureIgnore>
```

File: README.md [hunk 3]
```diff
@@ -156,6 +156,118 @@ By default, the setup wizard (Jenkins >= 2.0) is skipped when using `hpi:run`. I
...
+}
+```
+
+<!-- TODO when enough plugins have adopted this then remove this step -->
+Then set the following properties in your `pom.xml` to configure the Maven build to let Jenkins report the results:
+
+```xml
+<properties>
+  <maven.test.failure.ignore>false</maven.test.failure.ignore>
+  <testFailureIgnore>${maven.test.failure.ignore}</testFailureIgnore>
```
Comment (timja) [inline README.md:266]:
```suggestion
  <frontend.testFailureIgnore>${maven.test.failure.ignore}</frontend.testFailureIgnore>
```

File: pom.xml [hunk 4]
```diff
@@ -146,6 +146,8 @@
     <!-- may use e.g. -\-silent (without backslash) -->
     <npm.loglevel />
 
+    <testFailureIgnore>false</testFailureIgnore>
```
Comment (timja) [inline pom.xml:149]:
```suggestion
    <frontend.testFailureIgnore>false</frontend.testFailureIgnore>
```

File: pom.xml [hunk 5]
```diff
@@ -1184,7 +1186,7 @@
                 <configuration>
                   <!-- The package.json must define an "mvntest" script -->
                   <arguments>run mvntest</arguments>
-                  <testFailureIgnore>false</testFailureIgnore>
+                  <testFailureIgnore>${testFailureIgnore}</testFailureIgnore>
```
Comment (timja) [inline pom.xml:1189]:
```suggestion
                  <testFailureIgnore>${frontend.testFailureIgnore}</testFailureIgnore>
```

File: pom.xml [hunk 6]
```diff
@@ -1253,7 +1255,7 @@
...
```
Comment (timja) [inline pom.xml:1258]:
```suggestion
                  <testFailureIgnore>${frontend.testFailureIgnore}</testFailureIgnore>
```

**Pull request merged**